### PR TITLE
Operation/initial battery implementation

### LINKED ISSUE #157: Battery
This module should complete the following

- [ ] Collect telemetry 
- [ ] Pet watchdog

### PR DESCRIPTION
**Pull request by IsaacPoirier**
close #157 
Created RBattery.c and RBattery.h files. These files cover grabbing telemetry from the battery board (Voltage, Current, and Temperatures), setting the global variable safeFlag which is used for alerting when our battery is low on voltage (<6.5V), and also contains the object BatteryStatus which is used to store all of the aforementioned battery Telemetry.

### CODE REVIEW

File: radsat-sk/operation/subsystems/RBattery.c
Comment (ChrisAmaya) [inline radsat-sk/operation/subsystems/RBattery.c]: A similar comment was made in the eps PR. Review that PR first.

Wait until main cleanup is merged it will offer an example

File: radsat-sk/operation/subsystems/RBattery.c
Comment (ChrisAmaya) [inline radsat-sk/operation/subsystems/RBattery.c]: Same idea was mentioned in the PDB PR, you want to return the error and pass a pointer to a struct as a param instead

File: radsat-sk/operation/subsystems/RBattery.c
Comment (ChrisAmaya) [inline radsat-sk/operation/subsystems/RBattery.c]: implement an error check or else you are loading your struct with error codes instead of data

File: radsat-sk/operation/subsystems/RBattery.c
Comment (ChrisAmaya) [inline radsat-sk/operation/subsystems/RBattery.c]: need an error check here

File: radsat-sk/operation/subsystems/RBattery.c
Comment (ChrisAmaya) [inline radsat-sk/operation/subsystems/RBattery.c]: Changing the flag status is good enough and all this function needs to do

**Review by ChrisAmaya — DISMISSED** [review]

File: radsat-sk/operation/subsystems/RBattery.h
Comment (matthew-buglass) [inline radsat-sk/operation/subsystems/RBattery.h]: For error checking purposes and to stay consistent with other modules, I think the BatteryStatus object should be passed in as a pointer and an error code should be returned.

File: radsat-sk/operation/subsystems/RBattery.c
Comment (matthew-buglass) [inline radsat-sk/operation/subsystems/RBattery.c]: Should this return the flag as well to whoever is calling it so they don't need to look at it separately?

**Review by matthew-buglass — CHANGES_REQUESTED** [review]
Pretty much the same stuff Addi mentioned

File: radsat-sk/operation/subsystems/RBattery.c
Comment (tyrelkostyk) [inline radsat-sk/operation/subsystems/RBattery.c]: Style issue: public API must be prefixed with name of module. Also, getters typically don't need the word get as it's implied. Suggested rename: `batteryTelemetry()`

File: radsat-sk/operation/subsystems/RBattery.c
Comment (tyrelkostyk) [inline radsat-sk/operation/subsystems/RBattery.c]: Style issues.
1. i2c_command -> underscores not allowed; use `i2cCommand` instead. Could also just use `command` and `data`
2. Don't use "temp" in your variable names. How is someone supposed to know if it's a temporary variable, or a temperature variable? Our style guide prohibits most abbreviations

File: radsat-sk/operation/subsystems/RBattery.c
Comment (tyrelkostyk) [inline radsat-sk/operation/subsystems/RBattery.c]: Style guide: more whitespace. Change to `for (int i = 0; i < NUMBER_OF_VOLTAGE_AND_CURRENT_COMMANDS; i = i + 1) {` -> notice the whitespace around the curly braces and `for` keywords. This applies to all loops, `if` conditionals, etc.

File: radsat-sk/operation/subsystems/RBattery.c
Comment (tyrelkostyk) [inline radsat-sk/operation/subsystems/RBattery.c]: FYI you should be pulling in changes to `alpha` often.  To do that: `git pull && git merge origin/alpha`. You may have merge conflicts that you'll have to resolve.

Once you do that, you should do the following:
1. Add `#include <RCommon.h>`
2. Replace all `0` return codes with `SUCCESS`

File: radsat-sk/operation/subsystems/RBattery.c
Comment (tyrelkostyk) [inline radsat-sk/operation/subsystems/RBattery.c]: Style guide: change to `batteryIsSafe` or `batteryIsNotSafe` (whichever is more accurate) or similar

Comment (tyrelkostyk) [reply radsat-sk/operation/subsystems/RBattery.c]: Also, should this function offer a way for the calling API to know whether the flag was raised? Who is the inteded caller of this function?

Comment (tyrelkostyk) [reply radsat-sk/operation/subsystems/RBattery.c]: Also pt2: Update the docstring as it is outdated (claims no return value)

File: radsat-sk/operation/subsystems/RBattery.c
Comment (tyrelkostyk) [inline radsat-sk/operation/subsystems/RBattery.c]: Remove magic numbers; replace these with macros

File: radsat-sk/operation/subsystems/RBattery.c
Comment (tyrelkostyk) [inline radsat-sk/operation/subsystems/RBattery.c:331]: If an `if` condition returns, it does't need an else. Change to:

```c
if (...)
    return error;

return 0;
```

And of course see earlier comments regarding merging with alpha and using `SUCCESS` instead of `0`

File: radsat-sk/operation/subsystems/RBattery.h
Comment (tyrelkostyk) [inline radsat-sk/operation/subsystems/RBattery.h]: This needs to go at the very very bottom of the header

File: radsat-sk/operation/subsystems/RBattery.h
Comment (tyrelkostyk) [inline radsat-sk/operation/subsystems/RBattery.h]: Why is there a public global? If you want others to be able to view the status of the safeFlag, then provide a getter public API function e.g. `batterySafe()`. Move this to be a private variable in the source file.

File: radsat-sk/operation/subsystems/RBattery.h
Comment (tyrelkostyk) [inline radsat-sk/operation/subsystems/RBattery.h]: Types need to have a _t suffix, and lowercase with underscores structure. Change to `battery_status_t`

File: radsat-sk/operation/subsystems/RBattery.c
Comment (tyrelkostyk) [inline radsat-sk/operation/subsystems/RBattery.c]: If re-using a buffer for different purposes, you should make a habit of completely clearing the buffer beforehand.

e.g. add `memset(i2c_command, 0, sizeof(i2c_command));` before every `memcpy` line

Comment (tyrelkostyk) [reply radsat-sk/operation/subsystems/RBattery.c]: Same goes for data, you should clear it before passing it to another function (since it gets set by the talk function)

**Review by tyrelkostyk — CHANGES_REQUESTED** [review]
Overall:
- Style guide changes required
- Merge in alpha (see comments explaining how) and use updated return macros
- Refactor the safe flag stuff; make the variable private, and make the safe flag getter/setter function provide an API for returning the value

Otherwise it looks pretty good!

Let me know if you need any help with these suggestions.

File: radsat-sk/operation/subsystems/RBattery.c
Comment (matthew-buglass) [inline radsat-sk/operation/subsystems/RBattery.c:238]: Add documentation

File: radsat-sk/operation/subsystems/RBattery.c
Comment (matthew-buglass) [inline radsat-sk/operation/subsystems/RBattery.c:263]: Add documentation

**Review by matthew-buglass — DISMISSED** [review]
Just need to add some function documentation

File: radsat-sk/operation/subsystems/RBattery.c
Comment (tyrelkostyk) [inline radsat-sk/operation/subsystems/RBattery.c]: Should make this static, and prefix enums with "battery" i.e. `batteryCharging` and `batteryDischarging`, according to style guide. Also the anonymous name (`_batteryCurrentDirection`) should be `_battery_current_direction`. You could also typedef the enum if you wanted

File: radsat-sk/operation/subsystems/RBattery.c
Comment (tyrelkostyk) [inline radsat-sk/operation/subsystems/RBattery.c:181]: Might have to change the second argument to just `sizeof(storedData)` instead of `sizeof(&storedData[i])`. I think they're technically the same but the former is more clearer in my opinion

File: radsat-sk/operation/subsystems/RBattery.c
Comment (tyrelkostyk) [inline radsat-sk/operation/subsystems/RBattery.c]: What is 512? No magic numbers!

File: radsat-sk/operation/subsystems/RBattery.c
Comment (tyrelkostyk) [inline radsat-sk/operation/subsystems/RBattery.c]: 6.5?? A magic number? In MY code base?!

**Review by tyrelkostyk — APPROVED** [review]
Some small changes recommended

**Review by TheDopplerEffects — APPROVED** [review]
Looks like past reviews were resolved

**Review by TheDopplerEffects — APPROVED** [review]

**Pull request merged**
